tiledb/group.py

```diff
@@ -112,19 +113,19 @@ def __setitem__(self, key: str, value: GroupMetadataValueType):
 
             put_metadata = self._group._put_metadata
             if isinstance(value, np.ndarray):
-                put_metadata(f"{Group._NP_DATA_PREFIX}{key}", np.array(value))
-            elif isinstance(value, bytes):
-                put_metadata(key, lt.DataType.BLOB, len(value), value)
-            elif isinstance(value, str):
-                value = value.encode("UTF-8")
-                put_metadata(key, lt.DataType.STRING_UTF8, len(value), value)
-            elif isinstance(value, (list, tuple)):
-                put_metadata(key, np.array(value))
+                flat_value = value.ravel()
+                put_metadata(f"{Group._NP_DATA_PREFIX}{key}", flat_value)
+                if value.shape != flat_value.shape:
+                    self.__setitem__(f"{Group._NP_SHAPE_PREFIX}{key}", value.shape)
```

Comment:
Can you document what this code does here? If it is not a 1D array, then store the actual shape associated with the values? Also, does the code need to be called recursively or could it use `put_metadata`?

Reply:
Added more comments. We need the recursion because `value.shape` has to go through `pack_metadata_val` (avoiding code duplication). Since `value.shape` will always be [a tuple of ints](https://numpy.org/doc/2.0/reference/generated/numpy.shape.html), the else statement will be executed when `__setitem__` is given a numpy shape. I think this solution makes sense because the shape is like 'new' metadata.